colossalai/utils/model/pipelinable.py

```diff
@@ -0,0 +1,244 @@
+import torch
+import functools
+from colossalai.utils.model.utils import _substitute_init_recursively, InsertPostInitMethodToModuleSubClasses, call_to_str
+from colossalai.builder.pipeline import partition_uniform, partition_balanced
+from colossalai.core import global_context as gpc
+
+
+class Pipelinable(InsertPostInitMethodToModuleSubClasses):
+
+    def __init__(self):
+        super().__init__()
+        self._layer_spec_dict = {}
+        self._root_children = None
+        self._model = None
+        self._layer_spec_list = []
+        self._func_dict = {}
+        self._policy = "balanced"
+
+    @property
+    def policy(self):
+        return self._policy
+
+    @property
+    def layers_count(self):
+        return len(self._layer_spec_list)
+
+    @property
+    def funcs_count(self):
+        return len(self._func_dict)
+
+    def __enter__(self):
```

Comment:
It has been implemented in InsertPostInitMethodToModuleSubClasses. Why override?

Reply:
There is a tiny difference in the _post_init_method argument list. In the base class, _post_init_method only need module, but pipelinable need args and key world args to create the layer spec.

Reply:
add  args and key to InsertPostInitMethodToModuleSubClasses?

Reply:
Done.